diff -up dovecot-2.2.36/src/plugins/acl/acl-backend-vfile.c.aclfix dovecot-2.2.36/src/plugins/acl/acl-backend-vfile.c
--- dovecot-2.2.36/src/plugins/acl/acl-backend-vfile.c.aclfix	2018-09-18 15:00:08.778823903 +0200
+++ dovecot-2.2.36/src/plugins/acl/acl-backend-vfile.c	2018-09-18 15:00:08.814823737 +0200
@@ -161,8 +161,7 @@ acl_backend_vfile_object_init(struct acl
 	T_BEGIN {
 		if (*name == '\0' ||
 		    mailbox_list_is_valid_name(_backend->list, name, &error)) {
-			vname = *name == '\0' ? "" :
-				mailbox_list_get_vname(_backend->list, name);
+			vname = mailbox_list_get_vname(_backend->list, name);
 
 			dir = acl_backend_vfile_get_local_dir(_backend, name, vname);
 			aclobj->local_path = dir == NULL ? NULL :
